src/framework/constraints.ml

```diff
@@ -204,7 +213,7 @@ struct
 
   let startstate v = (S.startstate v, !start_level)
   let exitstate  v = (S.exitstate  v, !start_level)
-  let otherstate v = (S.otherstate v, !start_level)
+  (* let otherstate v = (S.otherstate v, !start_level) *)
```

Comment:
Why still keep this around here ?

Reply:
I now remember that there should be a TODO there because `threadenter` doesn't go back to `!start_level` right now. Whatever difference that makes...

Reply:
I guess that should be fixed then: The way I understand this lifter, it limits how deeply we descend into function calls. 
When we get below `0`, the `EvalFunvar` query will return `[]` and the call will be a nop.

In the case of a new thread, we should start this counter at `!start_level` again for the new thread, and not with the current one.